tests/taker_cli.rs

```diff
@@ -0,0 +1,255 @@
+use bitcoin::{address::NetworkChecked, Address, Amount, Transaction};
+use bitcoind::{bitcoincore_rpc::RpcApi, tempfile::env::temp_dir, BitcoinD, Conf};
+
+use std::{
+    fs,
+    path::{Path, PathBuf},
+    process::Command,
+    str::FromStr,
+};
+
+/// The taker-cli command struct
+/// Use it to perform all taker-cli operations
+struct TakerCli {
+    data_dir: PathBuf,
+    /// Bitcoind instance
+    bitcoind: BitcoinD,
+}
+
+impl TakerCli {
+    /// Construct a new [`TakerCli`] struct that also include initiating bitcoind.
+    fn new() -> TakerCli {
+        // Initiate the bitcoind backend.
+
+        let temp_dir = temp_dir().join(".coinswap");
+
+        // Remove if previously existing
+        if temp_dir.exists() {
+            fs::remove_dir_all::<PathBuf>(temp_dir.clone()).unwrap();
+        }
+
+        let mut conf = Conf::default();
+
+        conf.args.push("-txindex=1"); //txindex is must, or else wallet sync won't work.
+        conf.staticdir = Some(temp_dir.join(".bitcoin"));
+
+        log::info!("bitcoind configuration: {:?}", conf.args);
+
+        let os = std::env::consts::OS;
+        let arch = std::env::consts::ARCH;
+        let key = "BITCOIND_EXE";
+        let curr_dir_path = std::env::current_dir().unwrap();
+
+        let bitcoind_path = match (os, arch) {
+            ("macos", "aarch64") => curr_dir_path.join("bin").join("bitcoind_macos"),
+            _ => curr_dir_path.join("bin").join("bitcoind"),
+        };
+        std::env::set_var(key, bitcoind_path);
+        let exe_path = bitcoind::exe_path().unwrap();
+
+        log::info!("Executable path: {:?}", exe_path);
+
+        let bitcoind = BitcoinD::with_conf(exe_path, &conf).unwrap();
+
+        // Generate initial 101 blocks
+        let mining_address = bitcoind
+            .client
+            .get_new_address(None, None)
+            .unwrap()
+            .require_network(bitcoind::bitcoincore_rpc::bitcoin::Network::Regtest)
+            .unwrap();
+        bitcoind
+            .client
+            .generate_to_address(101, &mining_address)
+            .unwrap();
+
+        // derive data directory
+        let data_dir = temp_dir.join("taker");
+
+        TakerCli { data_dir, bitcoind }
+    }
+
+    // Build a cli-command
+    fn execute(&self, cmd: &[&str]) -> String {
+        let mut args = Vec::new();
+
+        // Data directory
+        args.push("--data-directory");
+        args.push(&self.data_dir.as_os_str().to_str().unwrap());
+
+        // Full node network
+        args.push("--NETWORK");
+        args.push("regtest");
+
+        // Connection network type (clearnet/tor)
+        args.push("--network");
+        args.push("clearnet");
+
+        // RPC authentication (user:password) from the cookie file
+        //
+        // get rpc_auth
+        // Path to the cookie file
```

Comment:
redundant docs? 

Tests in general don't need many docs. Docs should be only used to explain something that devs might miss while reading the code, or extra context/info.

Rust code is in general very verbose. doesn't need a lot of explainer doc. At tests, we should keep the docs minimal and only explain something that is not obvious from the code.